mapie/regression.py

```diff
@@ -479,9 +484,8 @@ def fit(
         cv = check_cv(self.cv)
         estimator = self._check_estimator(self.estimator)
         agg_function = self._check_agg_function(self.agg_function)
-        X, y = check_X_y(
-            X, y, force_all_finite=False, dtype=["float64", "int", "object"]
-        )
+        X, y = indexable(X, y)
```

Comment:
Same question here : we no longer check `X` ?

Reply:
Yes, check_array convert X to numpy array, which was the very cause of the bug. The user should be able to use any kind of X in input, provided its pipeline deals with it properly (strings, nan etc.)